activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/model/impl/ToSignalConverter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.runtime.api.model.impl;
+
+import org.activiti.api.runtime.model.impl.BPMNSignalImpl;
+import org.activiti.engine.delegate.event.ActivitiSignalEvent;
+import org.activiti.api.process.model.BPMNSignal;
+import org.activiti.api.process.model.payloads.SignalPayload;
+
+public class ToSignalConverter {
+
+    public BPMNSignal from(ActivitiSignalEvent internalEvent) {
+        BPMNSignalImpl signal = new BPMNSignalImpl(internalEvent.getActivityId(),
+                								   internalEvent.getActivityName(),
+                								   internalEvent.getActivityType());
+        
+        
+
+        signal.setProcessDefinitionId(internalEvent.getProcessDefinitionId());
+        signal.setProcessInstanceId(internalEvent.getProcessInstanceId());
+        signal.setSignalPayload((SignalPayload)internalEvent.getSignalData());
```

Comment:
This is not going to work as `internalEvent.getSignalData()` is never a `SignalPayload`. It's actually going to be variables. See https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/cmd/SignalEventReceivedCmd.java#L48 and https://github.com/Activiti/Activiti/blob/develop/activiti-engine/src/main/java/org/activiti/engine/impl/cmd/SignalEventReceivedCmd.java#L97-L99

Reply:
Have made changes and added a test. It seems that internal ActivitiSignalEvent does not have processInstanceId, but has a processDefinitionId... I don't know if this is a big problem....

Reply:
Thanks @CTI777. Ok, so `processInstanceId` is not set because you picked up a process with start signal event. For this specific test I think an intermediate catch signal event or a boundary signal event fits better. You can copy `ProcessWithBoundarySignal.bpmn20.xml` from runtime bundle starter, then `processInstanceId` will be set. Also, I'd recommend starting two instances of the process and verify in the test that two `SIGNAL_RECEIVED` are catch.

Reply:
@erdemedeiros 
I have added some more tests here and made some minor refactoring...